MyCryptoHQ/MyCrypto
# Reset Contracts-Interact tab on network change

Closes #2043 
Closes #1904 

### Description

Clears all fields in the Interact tab (contract name, address, and ABI/JSON interface) and closes the contract explorer, if open, when a user switches to a new network.

### Changes

InteractForm gets the current network as a prop from the Redux store. When the network changes, the fields are cleared and a new instance of the Dropdown component is created (otherwise, Dropdown's `selectedOption` will retain its previous value).

### Steps to Test

1. Go to the Contracts-Interact tab
2. Select a contract and click 'Access'
3. Select a different network in the top-right dropdown menu
4. Verify that all fields are cleared and the list of available contracts has been updated to reflect the new network
